### ISSUE TITLE
[Bug]: Playlist tab empty and corrupt playlists.db file

### ISSUE THREAD

**Issue by autonum** [FreeTubeApp/FreeTube#2802]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have searched the [issue tracker](https://github.com/FreeTubeApp/FreeTube/issues) for open and closed issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.

### Describe the bug

The playlist is shown empty.
The playlists.db file is still there and has a size of 2.8MB.

The error message appearing in the DevTools mentioning a corrupt file is attached.

![DevTools - corrupt db-playlists](https://user-images.githubusercontent.com/62657912/199506498-a9c69cf5-2e78-4a29-8107-a252b748d6bf.png)

I think it might have already been the case is the preceding version, as I notices the stars had turn to white (instead of yellow) in the subscription window; but I updated to the last version before having checked the «Playlist» tab.

Is there a way to repair such a corrupt file?

### Expected Behavior

Click on the «Playlist» tab and the content shows.

### Issue Labels

content not loading, data loss

### FreeTube Version

v0.18.0 Beta

### Operating System Version

Debian 11

### Installation Method

.deb

### Primary API used

Local API

### Last Known Working FreeTube Version (If Any)

v0.17.1 ?

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @autonum coukd u downgrade to V0.17.1 just to test if it is working on that version.

Also please add a title to your bug report.

**Comment by autonum** [comment]
Thanks for the quick reply.
How can I downgrade easily?
I can't find older versions on the website.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Ah yes, go to https://github.com/FreeTubeApp/FreeTube/releases/tag/v0.17.1-beta and u will find all the installers there.

**Comment by autonum** [comment]
When I select «Invidious API» instead of the local API, the error messages disappear.
I think that's where the problem came from.
But now the files are corrupted, they still won't load.

**Comment by autonum** [comment]
As I kept starting version 0.17.1, more and more error messages came on and now I just have a couple of channels left under the «Subscription» tap.
It seems that starting that version of FT damaged more and more of the profile files each time and left it in a very bad shape.
The 0.18.0 version gives no error message.
It would just need a way to repair the damaged files (playlists and profile).

**Comment by MarmadileManteater** [comment]
> As I kept starting version 0.17.1, more and more error messages came on and now I just have a couple of channels left under the «Subscription» tap. It seems that starting that version of FT damaged more and more of the profile files each time and left it in a very bad shape. The 0.18.0 version gives no error message. It would just need a way to repair the damaged files (playlists and profile).

This is a known issue. It should be fixed in 0.18.0.

**Comment by autonum** [comment]
Thank you.
Indeed, as mentioned, 0.18.0 seems to have this fixed.

Still, since the files were corrupted by 0.17.1, FT won't open them (safety reason, it says, to avoid data loss).

I think it must have affected the playlist database and the subscription one; since those two tabs are now (almost) empty.
Does anyone know which way(s) the files get corrupted and whether there is a mean to repair them?

Or do you mean that, in the 0.18.0 **not beta**, the corrupted files will become accessible again?

Have a good day.

**Comment by absidue** [comment]
The database files are just text, so you can open them in your favourite text editor and have a look at them. As for how the corruption happened, do you remember them breaking after specific things you did with FreeTube (might help work out where it's happening)? If you are comfortable sending the files here, we can take a look at them and see if we can fix the corruption for you. The fix might be as simple as deleting a few lines in them or more complicated, it's hard to say without seeing the files.

If you don't feel comfortable sending the files here, that's fine too, although you'll have to work out how to fix the files yourself.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
You could also reach out to us on IRC and private message us the files if u feel more comfortable with that

**Comment by autonum** [comment]
Hello.
Thank you very for taking the time to reply and offer to have a direct look at the files.

I've tried to find where the corruption had crept into the files in order to send you as much information as possible along with the damaged files but, not understanding the "normal" structure of the data, I could not see any clear evidence.
I just noticed that, when opened with Gedit with the «paragraph highlight» option, big bunches of the files are considered as one single line, then it "breaks" and another bunch is highlighted as one single line.
In the profiles database, it seems to correspond to a change in «channels profile» (the profiles that group different channels together). This might thus be the way it should be. (And, strangely, after looking through that file without modifying anything and starting FT again, all the channels – still checking – seem to be back in the «Subscription» tab. So this is one problem solved.)
But the «playlists» tab was still empty and the database file also shows these «paragraph breaks» although there should not be any reason for such breaks (if not after every line pointing to a single video, which is not the case).
Also, the file was fairly big (~3MB) for a few dozens videos.
Then I actually found out that the whole text had been copied more than 20(?) times (hence the line breaks, every time the whole thing would start again) and that was apparently (understandably) considered as corruption.

I then erased the whole file after the first paragraph and it all came back.

Conclusion :
For some reason, some API error kept on duplicating all the lines (as a whole) in the playlists.db file and, once above 10% corruption (which seems to have been set as the limit), it wouldn't open anymore.

As for the profiles.db, I don't know what happened but, after a long series of API error messages, I ended up with only two or three channels (out of ~50) showing in the «Subscription» tab. Then the problem solved itself, somehow, after opening the profiles.db file and closing it before starting FT again.

There's nothing else I can say about this case, except that I still get API error messages sometimes with the 0.18.0 version. Switching to «Local API» instead of «Invidious API» in the preferences seems to solve this.

Thanks again for you time and attention.
I hope this can helps others too.

Have a good day.
